site/content/blog/2025-03-14-hipcheck-3.12.0-release.md

```diff
@@ -0,0 +1,249 @@
+as pre-built binaries for our supported platforms, and they run as standalone
+programs except for their dependency on a sufficiently-recent version of libc
+from the system, or on CLI tools whose presence is checked during plugin
+configuration. This has worked well-enough for us, but until now we haven't
+had a decent answer for how to bundle and distribute plugins where you don't
+want to rely on the presence of dependencies on the host system.
+
+With 3.12.0, we've introduced new documentation for how to package and ship
+containerized plugins, including a new example found at
+`tests/test-plugins/activity-container` in the Hipcheck source repository,
+and we've updated the Rust Hipcheck SDK to make it easier to define a plugin
+which can run inside of a container.
+
+The key challenge for packaging a container-based plugin is making the CLI
+invocation for starting the plugin comply with Hipcheck's CLI protocol by
+accepting a `--port` flag to indicate what port the program should listen on.
+In our new example plugin, we do this with a shell script that wraps starting
+the plugin container and passess along the flag appropriately.
+
+Within the SDK, we also now expose an API for launching a plugin by having it
+listen on `0.0.0.0` instead of listening on the local loopback address
+`127.0.0.1`.
+
+We plan to continue to improve the story around shipping plugins with more
+complex dependencies, and to further improve documentation and tooling for this
+setup.
+
+## 🐍&nbsp;&nbsp;Experimental Python SDK for Plugins
+
+So far we've only shipped a plugin SDK for Rust, because Hipcheck itself is
+written in Rust and so we needed an SDK to use ourselves when we split our
+existing analyses out into plugins last year.
+
+However, the design of the plugin system is language-agnostic. So long as a
+program speaks the plugin protocol it can be used as a plugin.
+
+We're currently working on a plugin SDK for Python. It's still unfinished
+and experimental, but you can find it at `sdk/python` in the Hipcheck source
+repository, and we've reserved the package name `hipcheck-sdk` on PyPI, which
+we intend to start publishing to soon.
+
+If you're interested in developing a plugin in Python, feel free to try out
+the SDK. We'd love to hear feedback from people who do try it so we can address
+any deficiencies or awkwardness in the API or in the experience of shipping
+plugins written in a language that's not ahead-of-time compiled like Rust!
+
+## 🗺️&nbsp;&nbsp;How to Get Involved
+
+We're always looking for new contributors! If you'd like to learn more about
+Hipcheck and get involved in contributing, please checkout our
+[Roadmap][Roadmap] and feel free to get in touch with us through our
+[Discussions] board!
+
+## ⭐️&nbsp;&nbsp;Thank You to Our Contributors and Supporters
+
+As always, we want to say a big "Thank you!" to everyone who supports the
+project at MITRE, to CISA for [sponsoring our current work on it][cisa_hipcheck],
```

Comment:
It looks like there's no link reference (not sure of the official Markdown name for this) named `cisa_hipcheck`, so this renders without a link.